src/frequenz/channels/timer.py

```diff
@@ -605,7 +614,16 @@ def reset(self, *, start_delay: timedelta = timedelta(0)) -> None:
         if start_delay_ms < 0:
             raise ValueError(f"`start_delay` can't be negative, got {start_delay}")
         self._stopped = False
+
+        if interval is not None:
+            self._interval = _to_microseconds(interval)
+
+        prev_next_tick_time = self._next_tick_time
         self._next_tick_time = self._now() + start_delay_ms + self._interval
+
+        if prev_next_tick_time and prev_next_tick_time > self._next_tick_time:
```

Comment:
This check is to know if this is the first time the timer is started? I think it is worth a comment. Also for the `>` condition, as it wasn't obvious for me (I hope I'm getting it right).

```suggestion
        # If there was no previous `next_tick_time`, it means the timer wasn't started yet, so no need to
        # fire the reset event. If the new `_next_tick_time` is bigger than the previous one, we also don't
        # need to fire the reset even, as `ready()` will keep looping until the new `_next_tick_time` is
        # reached.
        if prev_next_tick_time and prev_next_tick_time > self._next_tick_time:
```

Reply:
Funny enough, I didn't even think about it, I was just making mypy happy :laughing:

Reply:
I was mainly checking if the new time is ealier or not like you described.
But I think we can simplify this.